CodeMaid/Logic/Cleaning/FileHeaderLogic.cs

```diff
@@ -58,16 +61,72 @@ internal void UpdateFileHeader(TextDocument textDocument)
                 return;
             }
 
+            settingsFileHeader = ReplaceVariables(textDocument, settingsFileHeader);
+
             var cursor = textDocument.StartPoint.CreateEditPoint();
             var existingFileHeader = cursor.GetText(settingsFileHeader.Length);
 
-            if (!existingFileHeader.StartsWith(settingsFileHeader))
+            if (!existingFileHeader.StartsWith(textDocument.GetCodeLanguageComment()))
             {
                 cursor.Insert(settingsFileHeader);
                 cursor.Insert(Environment.NewLine);
             }
         }
 
+        /// <summary>
+        /// Replace the variables in the header if any
+        /// </summary>
+        /// <param name="textDocument"></param>
+        internal string ReplaceVariables(TextDocument textDocument, string settingsFileHeader)
+        {
+            var outputSettingsFileHeader = settingsFileHeader;
+
+            foreach (var variable in _variables)
+            {
+                // If the file header doesn't contain this variable, do nothing...
+                if (!settingsFileHeader.Contains(variable))
+                {
+                    continue;
+                }
+
+                var variableValue = string.Empty;
+                switch (variable)
+                {
+                    case "$SOLUTION$":
+                        {
+                            if (_package.IDE.Solution != null)
+                            {
+                                variableValue = Path.GetFileNameWithoutExtension(_package.IDE.Solution.FullName);
+                            }
+
+                            break;
+                        }
+                    case "$PROJECT$":
+                        {
+                            try
+                            {
+                                variableValue = textDocument.Parent?.ProjectItem?.ProjectItems?.ContainingProject?.Name ?? "";
+                            }
+                            catch (Exception)
+                            {
```

Comment:
Please add some diagnostic logging when the containing project cannot be found.  Example:
```
OutputWindowHelper.DiagnosticWriteLine("Unable to retrieve containing project", ex);
```